lib/frames.js

```diff
@@ -26,17 +42,255 @@ frames.writeFrame = function(frame, stream) {
+    { name: 'target', type: '*', requires: terminus.Target },
+    { name: 'unsettled', type: 'map', default: {} },
+    { name: 'incompleteUnsettled', type: 'boolean', default: false },
+    { name: 'initialDeliveryCount', type: 'uint' },
+    { name: 'maxMessageSize', type: 'ulong', default: 0 },
+    { name: 'offeredCapabilities', type: 'symbol', multiple: true },
+    { name: 'desiredCapabilities', type: 'symbol', multiple: true },
+    { name: 'properties', type: 'fields', default: {} }
+  ]
+});
+
+frames.FlowFrame = defineFrame(FrameType.AMQP, {
+  name: 'flow', code: 0x13,
+  fields: [
+    { name:'nextIncomingId', type: 'uint' },
+    { name:'incomingWindow', type: 'uint', mandatory: true },
+    { name:'nextOutgoingId', type: 'uint', mandatory: true },
+    { name:'outgoingWindow', type: 'uint', mandatory: true },
+    { name:'handle', type: 'uint' },
+    { name:'deliveryCount', type: 'uint' },
+    { name:'linkCredit', type: 'uint' },
+    { name:'available', type: 'uint' },
+    { name:'drain', type: 'boolean', default: false },
+    { name:'echo', type: 'boolean', default: false },
+    { name:'properties', type: 'fields', default: {} }
+  ]
+});
+
+frames.TransferFrame = defineFrame(FrameType.AMQP, {
+  name: 'transfer', code: 0x14,
+  fields:[
+    { name: 'handle', type: 'uint', mandatory: true },
+    { name: 'deliveryId', type: 'uint' },
+    { name: 'deliveryTag', type: 'binary' },
+    { name: 'messageFormat', type: 'uint', default: 0 },
+    { name: 'settled', type: 'boolean' },
+    { name: 'more', type: 'boolean', default: false },
+    { name: 'rcvSettleMode', type: 'ubyte', default: 0 },
+    { name: 'state', type: 'delivery_state' },
+    { name: 'resume', type: 'boolean', default: false },
+    { name: 'aborted', type: 'boolean', default: false },
+    { name: 'batchable', type: 'boolean', default: false }
+  ]
+});
+
+frames.DispositionFrame = defineFrame(FrameType.AMQP, {
+  name: 'disposition', code: 0x15,
+  fields:[
+    { name: 'role', type: 'boolean', mandatory: true },
+    { name: 'first', type: 'uint', mandatory: true },
+    { name: 'last', type: 'uint' },
+    { name: 'settled', type: 'boolean', default: false },
+    { name: 'state', type: '*' },
+    { name: 'batchable', type: 'boolean', default: false }
+  ]
+});
+
+frames.DetachFrame = defineFrame(FrameType.AMQP, {
+  name: 'detach', code: 0x16,
+  fields: [
+    { name: 'handle', type: 'uint', mandatory: true },
+    { name: 'closed', type: 'boolean', default: false },
+    { name: 'error', type: 'error' }
+  ]
+});
+
+frames.EndFrame = defineFrame(FrameType.AMQP, {
+  name: 'end', code: 0x17,
+  fields: [
+    { name: 'error', type: 'error' }
+  ]
+});
+
+frames.CloseFrame = defineFrame(FrameType.AMQP, {
+  name: 'close', code: 0x18,
+  fields: [
+    { name: 'error', type: 'error' }
+  ]
+});
+
+// SASL frames
+frames.SaslMechanismsFrame = defineFrame(FrameType.SASL, {
+  name: 'sasl_mechanisms', code: 0x40,
```

Comment:
well we could go with snake_case for field names as well, but the general argument here is: field names are idiomatic names for properties that will be used by developers when working with a frame, whereas the name of the frame is used to create the symbolic descriptor for the composite type (though I do believe it should be `sasl-mechanisms` here instead of using the _)